Mathlib/Geometry/Manifold/ChartedSpace.lean

```diff
@@ -978,6 +978,18 @@ instance hasGroupoid_continuousGroupoid : HasGroupoid M (continuousGroupoid H) :
   simp only [and_self_iff]
 #align has_groupoid_continuous_groupoid hasGroupoid_continuousGroupoid
 
+/-- If `G` is closed under restriction, the transition function between
+  the restriction of two charts `e` and `e'` lies in `G`. -/
+theorem StructureGroupoid.trans_restricted {e e' : PartialHomeomorph M H} {G : StructureGroupoid H}
+    (he : e ∈ atlas H M) (he' : e' ∈ atlas H M)
+    [HasGroupoid M G] [ClosedUnderRestriction G] (s : Opens M) [Nonempty s] :
```

Comment:
Is it helpful to invoke type class inference in `[Nonempty s]`? `s` is some subset, which in general is unlikely to have an instance. My hunch is to write instead `(hs : Set.Nonempty s)`.

Reply:
Well, `subtypeRestr` requires an argument `[Nonempty s]`, which is why I'm also using this form...

Reply:
My doubts extend to `subtypeRestr`. In `ChartedSpace.lean`, the only place so far where it's used, you see `Nonempty` instance needing to be constructed manually, such as `@PartialHomeomorph.subtypeRestr _ _ _ _ (chartAt H x.1) s ⟨x⟩` in `[TopologicalSpace.Opens.instChartedSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/ChartedSpace.html#TopologicalSpace.Opens.instChartedSpace)` and `haveI : Nonempty U := ⟨x⟩` in [TopologicalSpace.Opens.chartAt_inclusion_symm_eventuallyEq](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/ChartedSpace.html#TopologicalSpace.Opens.chartAt_inclusion_symm_eventuallyEq). I think I'll open a PR changing the instance argument to and explicit one.

Reply:
Sounds good. If that PR is approved, I'll update mine accordingly.